### PR TITLE
Use libvirt API for migration, not shell commands

### PR DESCRIPTION
**Pull request by TuxPowered42**

### CODE REVIEW

File: igvm/kvm.py [hunk 1]
```diff
@@ -243,14 +243,70 @@ def migrate_live(source, destination, vm, domain):
...
-            domain=domain.name(),
-            destination=destination.fqdn,
-        ),
-        with_sudo=False,
+    migrate_flags = (
+        VIR_MIGRATE_LIVE |  # Do it live
+        VIR_MIGRATE_PERSIST_DEST |  # Define the VM on the new host
+        VIR_MIGRATE_CHANGE_PROTECTION |  # Protect source VM
+        VIR_MIGRATE_NON_SHARED_DISK |  # Copy non-shared storage
+        VIR_MIGRATE_AUTO_CONVERGE |  # Slow down VM if can't migrate memory
```
Comment (kofrezo) [inline igvm/kvm.py]: What is the effect here ? How does it "slow down" the VM ?

File: igvm/kvm.py [hunk 2]
```diff
@@ -243,14 +243,70 @@ def migrate_live(source, destination, vm, domain):
...
+                log.info('Waiting for migration stats to show up')
+            time.sleep(1)
+    except KeyboardInterrupt:
+        log.warning('Cancelling migration')
+        domain.abortJob()
+
+    log.info('Awaiting migration to finish')
+    try:
+        future.result()
+    except Exception as e:
```
Comment (kofrezo) [inline igvm/kvm.py]: Do we have a more specific Exception we can catch ?

**Comment by HerrSpace** [comment]
Looks good.

File: igvm/kvm.py (hunk 1 above)
Comment (TuxPowered42) [reply igvm/kvm.py]: Also this is the exact copy of the old parameters. We might change them to post-copy migration for performance reasons later.

**Pull request merged**